domainservice/src/main/kotlin/systems/ajax/malov/stockanalyzer/kafka/producer/StockPriceNotificationProducer.kt

```diff
@@ -0,0 +1,43 @@
+package systems.ajax.malov.stockanalyzer.kafka.producer
+
+import org.apache.kafka.clients.producer.ProducerRecord
+import org.springframework.stereotype.Component
+import reactor.core.publisher.Mono
+import reactor.kafka.sender.KafkaSender
+import reactor.kafka.sender.SenderRecord
+import reactor.kotlin.core.publisher.toFlux
+import reactor.kotlin.core.publisher.toMono
+import systems.ajax.malov.internalapi.KafkaTopic
+import systems.ajax.malov.internalapi.output.pubsub.stock.notification_stock_price.proto.NotificationStockPrice
+import java.time.Duration
+
+@Component
+class StockPriceNotificationProducer(
+    private val kafkaNotificationStockPriceKafkaProducer: KafkaSender<String, NotificationStockPrice>,
+) {
+    fun sendNotificationStockPrice(notificationStockPrices: List<NotificationStockPrice>): Mono<Unit> {
+        return notificationStockPrices
+            .toFlux()
+            .buffer(Duration.ofSeconds(1))
```

Comment:
what the logic behind here? you create flux without any blocking operations (instantly) and for some reason you want to buffer it to list again with 1s delay

why can't you just pass list to producer?